cloudinary-core/src/main/java/com/cloudinary/provisioning/Account.java

```diff
@@ -0,0 +1,218 @@
+package com.cloudinary.provisioning;
+
+import com.cloudinary.Api;
+import com.cloudinary.Cloudinary;
+import com.cloudinary.api.ApiResponse;
+import com.cloudinary.utils.ObjectUtils;
+
+import java.util.*;
+
+public class Account {
+    private static final String CLOUDINARY_ACCOUNT_URL = "CLOUDINARY_ACCOUNT_URL";
+    private static final String SUB_ACCOUNTS = "sub_accounts";
+    private static final String USERS = "users";
+    private static final String USER_GROUPS = "user_groups";
+
+    private final AccountConfiguration configuration;
+    private final String PROVISIONING = "provisioning";
+    private final String ACCOUNTS = "accounts";
+    private final String accountId;
+    private final String key;
+    private final String secret;
+    private final Api api;
+
+    public Account(Cloudinary cloudinary) {
+        String provisioningData = System.getProperty(CLOUDINARY_ACCOUNT_URL, System.getenv(CLOUDINARY_ACCOUNT_URL));
+        if (provisioningData != null) {
+            this.configuration = AccountConfiguration.from(provisioningData);
+            this.accountId = configuration.accountId;
+            this.key = configuration.provisioningApiKey;
+            this.secret = configuration.provisioningApiSecret;
+        } else {
+            throw new IllegalArgumentException("Must provide configuration instance or set an ENV variable: " +
+                    "CLOUDINARY_ACCOUNT_URL=account://provisioning_api_key:provisioning_api_secret@account_id");
+        }
+        
+        this.api = cloudinary.api();
+    }
+
+    public Account(AccountConfiguration accountConfiguration, Cloudinary cloudinary) {
+        this.configuration = accountConfiguration;
+        this.api = cloudinary.api();
+        this.accountId = accountConfiguration.accountId;
+        this.key = accountConfiguration.provisioningApiKey;
+        this.secret = accountConfiguration.provisioningApiSecret;
+    }
+
+    private ApiResponse callAccountApi(Api.HttpMethod method, List<String> uri, Map<String, Object> params, Map<String, Object> options) throws Exception {
+        options = verifyOptions(options);
+
+        if (!options.containsKey("provisioning_api_key")) {
```

Comment:
We can't mix and match, it's probably better to end up having just the key if the secret is missing, otherwise we end up in an inconsistent state. Let's discuss it tomorrow to go over all the scenarios...

Reply:
I know we can't. that's why I suggest verify that both were sent and react if not

Reply:
But then if we support key only API, there's no way to override the config secret (e.g. send null in the options, which will only be used if the key is there).

If we can safely assume we don't have key only APIs (only uploader has some of those) then we should indeed verify both are in the options.